internal/blocksync/reactor.go

```diff
@@ -601,7 +601,11 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 			r.pool.PopRequest()
 
 			// TODO: batch saves so we do not persist to disk every block
-			r.store.SaveBlock(first, firstParts, extCommit)
+			if state.ConsensusParams.ABCI.VoteExtensionsEnabled(state.LastBlockHeight) {
+				r.store.SaveBlockWithExtendedCommit(first, firstParts, extCommit)
+			} else {
+				r.store.SaveBlock(first, firstParts, extCommit.ToCommit())
```

Comment:
So avoid unexpected side-effects, I would play on the safe side here and would use `second`'s `LastCommit`.
Remember that, before all our changes, the block store's `SaveBlock` was always fed a _canonical_ commit (which `extCommit` isn't).
Not a big deal, but would help me sleep the night :-)